File: backend/app/schemas/document.py
from datetime import datetime

from pydantic import BaseModel, Field


class DocumentResponse(BaseModel):
    id: str
    attachment_id: str | None
    name: str
    mime_type: str
    size_bytes: int
    status: str
    error_message: str | None
    created_at: datetime
    updated_at: datetime

    model_config = {
        "from_attributes": True,
    }


class DocumentChunkResponse(BaseModel):
    id: str
    chunk_index: int
    content: str
    character_count: int

    model_config = {
        "from_attributes": True,
    }


class DocumentDetailResponse(
    DocumentResponse
):
    chunks: list[
        DocumentChunkResponse
    ] = Field(
        default_factory=list
    )

File: frontend/src/types/document.ts
export interface DocumentChunk {
  id: string;
  chunk_index: number;
  content: string;
  character_count: number;
}

export interface IndexedDocument {
  id: string;
  attachment_id: string | null;
  name: string;
  mime_type: string;
  size_bytes: number;
  status: "processing" | "ready" | "failed";
  error_message: string | null;
  created_at: string;
  updated_at: string;
}

export interface IndexedDocumentDetail
  extends IndexedDocument {
  chunks: DocumentChunk[];
}